### PR TITLE
Change delimiter enhancement for HUGO

### PR DESCRIPTION
**Pull request by danke-click**
*DelimiterLeft* and *DelimiterRight* can be set in config file. Default is "{{" and "}}".

### CODE REVIEW

**Comment by bep** [comment]
Thanks for this. As you might notice, there are plenty test failures.

Some general comments:

I havent read too carefully through your changes, but:

1) Changes in 7 files tells me that this can be done much simpler. My initial thought when I looked at it was to hook into the AddInternalTemplate or something: https://github.com/spf13/hugo/blob/be2097e1ad789eca5d893805a059d94defbe5c48/tpl/template.go#L139
2) Your commit log looks messy and none of your commit messages ends with the magic `Fixes #1044` A good resource for quality commit messages: http://chris.beams.io/posts/git-commit/

File: helpers/content.go [hunk 1]
```diff
@@ -367,3 +367,18 @@ func GetRstContent(content []byte) string {
...
+// Reads the Delimiter Settings from configuration
+// Default is DelimiterLeft = "{{" and DelimiterRight = "}}"
+var DLeft string
+var DRight string
+var ReplaceDelimiters bool = (DLeft != "{{")
+
+// Replaces the delimiters in hardcoded internal templates
+func SetDelims(snippet string) string {
+	if ReplaceDelimiters {
+		snippet = strings.Replace(snippet, "{{", DLeft, -1)
```
Comment (bep) [inline helpers/content.go]: strings.Replacer would be better in this case.

File: helpers/content.go [hunk 2]
```diff
@@ -367,3 +367,18 @@ func GetRstContent(content []byte) string {
...
 }
+
+// Reads the Delimiter Settings from configuration
+// Default is DelimiterLeft = "{{" and DelimiterRight = "}}"
+var DLeft string
+var DRight string
+var ReplaceDelimiters bool = (DLeft != "{{")
+
+// Replaces the delimiters in hardcoded internal templates
+func SetDelims(snippet string) string {
```
Comment (bep) [inline helpers/content.go]: This func belongs in the tpl package and should be named more descriptive.

**Comment by danke-click** [comment]
*Thank you very much* for your Feedback, your comments and all the additional information that you gave me. I really appreciate it! I also agree with your recommendations and already changed a lot of the code, but it will take more time to figure out how to deal with github in a good quality way ...

**Important:** When I analyzed the code I found several code blocks that generate pure HTML Code or files without using the template package. For example in *htmlredirect.go* to create aliases and there are other places as well.

For that reason I think we still need a helper function to correct the delimiters in that code. So I like to keep the function in the helper package, if that is okay? Otherwise I am totaly open to any suggestion to solve this problem.

I will come back, as soon I find out how to do the ```rebase``` thing :), 
thanks again, marc

**Comment by bep** [comment]
1) None of your changes tells me that the SetDelim method will have to live outside the tpl package, and if so, it should have a different name (you lose the package context). No need to "fix" the templates in the tests. I would, however, expect to see some test cases that loop over both the regular case and some others to verify that the end result is the same.
2) For rebase, this resource may help: http://discuss.gohugo.io/t/howto-rebase-your-feature-branch/442/3

**Comment by spf13** [comment]
@danke-click How is this coming along? Do you need any more help?

**Comment by bep** [comment]
@danke-click any updates on this?

**Comment by danke-click** [comment]
No. I've made all the changes, including all your recommendations a long
time ago.

Problem is, I don't have time to learn github, how to maintain a clear
commit log etc.  -  sorry, but I don't think I'll send any updates anymore.

From my point of view it is faar tooo complicated to deal with a system
like github if you don't use it regularly.



Am 24.07.2015 um 11:46 schrieb Bjørn Erik Pedersen:

**Comment by bep** [comment]
Replaced by https://github.com/spf13/hugo/issues/1300

**Pull request closed without merging**